pkg/sql/resolver.go

```diff
@@ -299,69 +299,61 @@ func (p *planner) IsTypeVisible(
 	return false, true, nil
 }
 
-// HasPrivilege is part of the tree.EvalDatabase interface.
-func (p *planner) HasPrivilege(
+// HasAnyPrivilege is part of the tree.EvalDatabase interface.
+func (p *planner) HasAnyPrivilege(
 	ctx context.Context,
 	specifier tree.HasPrivilegeSpecifier,
 	user security.SQLUsername,
-	priv privilege.Privilege,
-) (bool, error) {
+	privs []privilege.Privilege,
```

Comment:
Checks multiple privileges now to avoid multiple calls to `ResolveDescriptorForPrivilegeSpecifier`.

Returns `tree.HasAnyPrivilegeResult` to avoid needing to unwrap and inspect error pg codes at call sites.